plugins/inputs/win_wmi/win_wmi.go

```diff
@@ -0,0 +1,268 @@
+	}
+	return tagFilter, nil
+}
+
+func BuildWqlStatements(s *Wmi) {
+	for i, q := range s.Queries {
+		s.Queries[i].Query = BuildWqlStatement(q)
+	}
+}
+
+// build a WMI query from input configuration
+func BuildWqlStatement(q Query) string {
+	wql := fmt.Sprintf("SELECT %s FROM %s", strings.Join(q.Properties, ", "), q.ClassName)
+	if len(q.Filter) > 0 {
+		wql = fmt.Sprintf("%s WHERE %s", wql, q.Filter)
+	}
+	return wql
+}
+
+func DoQuery(q Query, acc telegraf.Accumulator) error {
+	// The only way to run WMI queries in parallel while being thread-safe is to
+	// ensure the CoInitialize[Ex]() call is bound to its current OS thread.
+	// Otherwise, attempting to initialize and run parallel queries across
+	// goroutines will result in protected memory errors.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	tags := map[string]string{}
+	fields := map[string]interface{}{}
+
+	// init COM
+	if err := ole.CoInitializeEx(0, ole.COINIT_MULTITHREADED); err != nil {
+		oleCode := err.(*ole.OleError).Code()
+		if oleCode != ole.S_OK && oleCode != sFalse {
+			return err
+		}
+	}
+	defer ole.CoUninitialize()
+
+	unknown, err := oleutil.CreateObject("WbemScripting.SWbemLocator")
+	if err != nil {
+		return err
+	} else if unknown == nil {
+		return fmt.Errorf("failed to create WbemScripting.SWbemLocator, is WMI broken?")
+	}
+	defer unknown.Release()
+
+	wmi, err := unknown.QueryInterface(ole.IID_IDispatch)
+	if err != nil {
+		return fmt.Errorf("failed to QueryInterface: %v", err)
+	}
+	defer wmi.Release()
+
+	// service is a SWbemServices
+	serviceRaw, err := oleutil.CallMethod(wmi, "ConnectServer", nil, q.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed calling method ConnectServer: %v", err)
+	}
+	service := serviceRaw.ToIDispatch()
+	defer func() { _ = serviceRaw.Clear() }()
+
+	// result is a SWBemObjectSet
+	resultRaw, err := oleutil.CallMethod(service, "ExecQuery", q.Query)
+	if err != nil {
+		return fmt.Errorf("failed calling method ExecQuery for query %s: %v", q.Query, err)
+	}
+	result := resultRaw.ToIDispatch()
+	defer func() { _ = resultRaw.Clear() }()
+
+	count, err := oleInt64(result, "Count")
+	if err != nil {
+		return fmt.Errorf("failed getting Count: %v", err)
+	}
+
+	for i := int64(0); i < count; i++ {
+		// item is a SWbemObject
+		itemRaw, err := oleutil.CallMethod(result, "ItemIndex", i)
+		if err != nil {
+			return fmt.Errorf("failed calling method ItemIndex: %v", err)
+		}
+
+		outerErr := func() error {
+			item := itemRaw.ToIDispatch()
+			defer item.Release()
+
+			for _, wmiProperty := range q.Properties {
+				innerErr := func() error {
+					prop, err := oleutil.GetProperty(item, wmiProperty)
+					if err != nil {
+						return fmt.Errorf("failed GetProperty: %v", err)
+					}
+					defer func() { _ = prop.Clear() }()
+
+					// if an empty property is returned from WMI, then move on
+					if prop.Value() == nil {
+						return nil
+					}
```

Comment:
I don't think this should be done? As the Property is requested explicitly in the config.

Reply:
I updated the plugin to allow null properties as results.